pccommon/pccommon/middleware.py

```diff
@@ -1,96 +1,99 @@
 import asyncio
 import logging
 import time
-from typing import Awaitable, Callable
+from functools import wraps
+from typing import Any, Callable
 
-from fastapi import HTTPException, Request, Response
+from fastapi import HTTPException, Request
 from fastapi.applications import FastAPI
-from starlette.middleware.base import BaseHTTPMiddleware
-from starlette.responses import PlainTextResponse
+from fastapi.dependencies.utils import (
+    get_body_field,
+    get_dependant,
+    get_parameterless_sub_dependant,
+)
+from fastapi.responses import PlainTextResponse
+from fastapi.routing import APIRoute, request_response
 from starlette.status import HTTP_504_GATEWAY_TIMEOUT
-from starlette.types import Message
 
 from pccommon.logging import get_custom_dimensions
-from pccommon.tracing import trace_request
 
 logger = logging.getLogger(__name__)
 
 
-async def handle_exceptions(
-    request: Request,
-    call_next: Callable[[Request], Awaitable[Response]],
-) -> Response:
-    try:
-        return await call_next(request)
-    except HTTPException:
+async def http_exception_handler(request: Request, exc: Exception) -> Any:
+    # Log the exception with additional request info if needed
+    logger.exception("Exception when handling request", exc_info=exc)
```

Comment:
Seems like this will result in two mostly identical logs, and there's not much differentiation between the conditional check below (we'll raise and log an exception, except twice in the else)

Reply:
This is something we still want to resolve -- let me know if the above comment isn't clear.

Reply:
Now that this is back, I think we still have to resolve this double logging when it's an HTTPException. Since we don't do anything different in the conditional (we log and raise, each time) we can probably just simply have this be a single log and raise. I tested the `logger.exception("Exception when handling request", exc_info=exc)` style and this ended up in the exception table with the stack trace and everything we'd want in the custom dimensions. I'd recommend this whole function just be

```python
logger.exception("Exception when handling request", exc_info=exc)
raise
```